lua/lazyvim/plugins/extras/lang/python.lua

```diff
@@ -0,0 +1,110 @@
+return {
```

Comment:
The virtual env doesn't have to live under `.venv/`.
`Pipenv` and `conda` for example put them in designated directories outside of your repo.

I've been using something like this to start my language server with the correct python bin.
```suggestion
local lsp_path = require("lspconfig/util").path

local function is_empty(s)
  return s == nil or s == ""
end

local function get_pipenv_venv_path(workspace)
  workspace = workspace or vim.fn.getcwd()

  -- Find and use virtualenv from pipenv in workspace directory.
  local match = vim.fn.glob(lsp_path.join(workspace, "Pipfile"))
  if not is_empty(match) then
    local venv = vim.fn.trim(vim.fn.system("PIPENV_PIPFILE=" .. match .. " pipenv --venv 2>/dev/null"))
    vim.notify("Found pipenv venv", "debug", { title = "Python" })
    return venv
  end
  return ""
end

local function get_peotry_venv_path(workspace)
  workspace = workspace or vim.fn.getcwd()

  -- Find and use virtualenv via poetry in workspace directory.
  local match = vim.fn.glob(lsp_path.join(workspace, "poetry.lock"))
  if not is_empty(match) then
    local venv = vim.fn.trim(vim.fn.system("poetry env info -p 2>/dev/null"))
    vim.notify("Found poetry venv", "debug", { title = "Python" })
    return venv
  end
  return ""
end

local function get_local_venv_path(workspace)
  local path = lsp_path.join(workspace, ".venv")
  if not is_empty(vim.fn.glob(path)) then
    vim.notify("Found local venv", "debug", { title = "Python" })
    return lsp_path.dirname(path)
  end

  return ""
end

local function get_venv_path(workspace)
  workspace = workspace or vim.fn.getcwd()
  local venv = vim.VIRTUAL_ENV
    or get_pipenv_venv_path(workspace)
    or get_peotry_venv_path(workspace)
    or get_local_venv_path(workspace)

  vim.notify("Using  venv: " .. venv, "debug", { title = "Python" })
  return venv
end

return {
```

Use it later as:
```python
      vim.list_extend(opts.sources, {
        -- Order of formatters matters. They are used in order of appearance.
        nls.builtins.formatting.ruff.with({
          filetypes = { "python" },
          prefer_local = lsp_path.join(get_venv_path(), "bin"),
        }),
        nls.builtins.formatting.black.with({
          filetypes = { "python" },
          prefer_local = lsp_path.join(get_venv_path(), "bin"),
        }),
      })
-- ...
    opts = {
      servers = {
        pyright = {
          before_init = function(_, config)
            local path = lsp_path.join(get_venv_path(), "bin", "python")
            config.settings.python.pythonPath = path
          end,
          settings= {...}

```

Reply:
`pyright` respects the `pyproject.toml` or `pyrightconfig.json` configuration files. You can also set the python path in these files. I do not plan to add automatic detection to this extra as it will complicate a lot.